modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/MultiStatementHandler.java

```diff
@@ -141,14 +143,31 @@ CompletableFuture<AsyncSqlCursor<InternalSqlRow>> processNext() {
                                 new IteratorToDataCursorAdapter<>(Collections.emptyIterator()),
                                 nextCurFut
                         ));
+            } else if (parsedResult.queryType() == SqlQueryType.DDL) {
+                List<ParsedResultWithNextCursorFuture> ddlBatch = new ArrayList<>();
+
+                do {
+                    ddlBatch.add(new ParsedResultWithNextCursorFuture(scriptStatement.parsedResult, scriptStatement.nextStatementFuture));
+
+                    ScriptStatement statement = statements.peek();
+                    if (statement == null || statement.parsedResult.queryType() != SqlQueryType.DDL) {
+                        break;
+                    }
+
+                    scriptStatement = statement;
+
+                    statements.poll();
+                } while (true);
+
+                fut = query.executor.executeChildBatch(query, scriptTxContext, statementNum, ddlBatch);
             } else {
                 scriptTxContext.registerCursorFuture(parsedResult.queryType(), cursorFuture);
 
                 fut = query.executor.executeChildQuery(query, scriptTxContext, statementNum, parsedResult, params, nextCurFut);
             }
 
             boolean implicitTx = scriptTxContext.explicitTx() == null;
-
+            ScriptStatement lastProcessedStatement = scriptStatement;
```

Comment:
maybe something like this would be better?

```suggestion
            boolean isLastScriptStatement = lastProcessedStatement.isLastStatement()
```

Reply:
replaced, thanks for suggestion